netket/operator/_abstract_operator.py

```diff
@@ -199,8 +199,14 @@ def get_conn(self, x):
         Returns:
             matrix: The connected states x' of shape (N_connected,hilbert.size)
             array: An array containing the matrix elements :math:`O(x,x')` associated to each x'.
-
+        
+        Raise:
+            ValueError: If the given quantum number is not compatible with the hilbert space.
         """
+        if x.size != self.hilbert.size:
```

Comment:
shouldn't we check the shape of x as well?

Reply:
Sure we can do. I just ignored it since we have x.reshape a few lines below anyway. But you are right, the error might be misleading otherwise.

Reply:
Yeah, maybe just check the shape to be (N,)